### PR TITLE
deprecate the UNITTYPE_ constants, and the commands that use them

### LINKED ISSUE #1677: deprecate UNITTYPE_ constants, switch to hardcoded BL_ constants
For the sake of simplicity I believe we should deprecate the `UNITTYPE_` constants so we only have the `BL_` constants.
* The `BL_` constants have the same values as in the `bl_type` enum
* The `BL_` constants can be bitmasked, enabling to pass more than one type to a command
* The `BL_` constants are hardcoded in script.c so if the enum ever changes they will still be up-to-date

### PR DESCRIPTION
**Pull request by Helianthella**
If we make `getmapxy()` accept `BL_` constants then we can't make it backwards-compatible with `UNITTYPE_` constants, because if an integer is given instead of a constant there's no way to know if the value is intended to be one of `UNITTYPE_` or `BL_`, so it's better to just deprecate the whole command altogether, since it's already integrated into `getunitdata()`. Also `getmapxy()` has some weird behaviour, such as finding the location of a pet by the name of its owner (WTF?) and is quite messy in general.

closes #1677 

<br> <br> <br>

- - -

<sub>BTW this is meko, I changed by username</sub>

### CODE REVIEW

File: npc/other/Global_Functions.txt [hunk 1]
```diff
@@ -438,3 +438,13 @@ function	script	F_ShuffleNumbers	{
 function	script	F_MesColor	{
 	return sprintf("^%06X", min(getarg(0), 0xFFFFFF));
 }
+
+// getunitlocation(<unit id>, <map variable>, <x variable>, <y variable>)
+//     syntax sugar for getunitdata
+function	script	getunitlocation	{
```
Comment (Helianthella) [inline npc/other/Global_Functions.txt]: this is a drop-in replacement for `getmapxy()` for old scripts

**Comment by Emistry** [comment]
why not just fix and update the `getmapxy` instead? from a 1 line of codes it has now basically become 5+ lines... and lost some of the feature from `getmapxy` for example it able to check if the server able to retrieve the unit location. Of course it could be done for the new `getunitlocation()` but why create so much hassle just for a changes like this? It has no backward-compatibility for older Herc too.

anyway
```
function	script	getunitlocation	{
```
should have be
```
function	script	F_GetUnitLocation	{
```

**Comment by Helianthella** [comment]
@emistry it is not possible to make `getmapxy` support `BL_` constants without breaking backward-compatibility. there is simply no easy fix

**Comment by dastgirp** [comment]
I believe deprecating getmapxy would be bad, since it's used even in old scripts (since eA) and updating everything would be troublesome. Or might be not as handy.

Can there be some kind of processing to detect if BL_ constant was used or UNITTYPE_ was used and throw deprecation accordingly

**Comment by Helianthella** [comment]
@dastgirp yes, there can, but as stated in the first post if a scripter decides to pass a number instead of a constant then there's no way to know which enum it belongs to

**Comment by Helianthella** [comment]
this would cause too much chaos, so I'm closing it

**Pull request closed without merging**